### PR TITLE
Feat: linear integration

### LINKED ISSUE #231: [frontend] profile page ui improvment & linear integration
enable user to integrate linear with march from profile page

### PR DESCRIPTION
**Pull request by hanzili**
#231 
should merge #241 before this one
## second part of this issue
- create `useLinear` hook to interact with Linear related apis
- interact with backend to complete linear integration

Screenshot 2024-10-07 at 21 49 02
Screenshot 2024-10-07 at 21 49 27

### CODE REVIEW

**Comment by madhav-relish** [comment]
Update the .env,example and also add instructions for setting up linear for developers @hanzili

**Comment by hanzili** [comment]
> Update the .env,example and also add instructions for setting up linear for developers @hanzili

Done! I think we should add more integration instruction (for google, calendar etc.) to README

File: README.md [hunk 1]
```diff
@@ -46,3 +45,18 @@ Go to the `app/backend` and `app/frontend` directories and duplicate the `.env.e
...
+### Linear integration
+
+To set up the Linear integration:
+1. Sign up for a Linear account at https://linear.app if you haven't already.
+2. Create a new Linear OAuth application in your Linear settings.
+3. Fill in the following environment variables in your `.env` file:
+
+```
+LINEAR_REDIRECT_URL=http://localhost:3000/api/auth/callback/linear
+LINEAR_CLIENT_ID=<your_linear_client_id>
```
Comment (madhav-relish) [inline README.md:61]: This should be LINEAR_REDIRECT_URL=http://localhost:3000/auth/linear

File: apps/frontend/src/hooks/useLinear.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { useCallback, useState } from "react"
+import axios from "axios"
+import { useAuth } from "../contexts/AuthContext"
+import { BACKEND_URL } from "../lib/constants/urls"
+
+interface LinearIssue {}
+
```
Comment (madhav-relish) [inline apps/frontend/src/hooks/useLinear.ts:36]: Can also add properties here, rather than using an empty object

**Review by madhav-relish — COMMENTED** [review]
Everything else looks good!

**Comment by sajdakabir** [comment]
Hey @hanzili, could you please pull from the main branch and also fix the build error? To check the build error, please use the command: pnpm run build


Screenshot 2024-10-10 at 11 00 35 PM

**Comment by hanzili** [comment]
> Hey @hanzili, could you please pull from the main branch and also fix the build error? To check the build error, please use the command: pnpm run build
> 
> Screenshot 2024-10-10 at 11 00 35 PM

Done!
I ran `pnpm run format` so some other files are also changed

**Comment by sajdakabir** [comment]
@hanzili , screenshot is not visible. Let me check on my local.

**Comment by hanzili** [comment]
> @hanzili , screenshot is not visible. Let me check on my local.

You mean which screenshot? :0

**Comment by immadsahin** [comment]
@hanzili  the screenshot you posted in the comment

**Comment by hanzili** [comment]
> @hanzili the screenshot you posted in the comment

Screenshot 2024-10-11 at 00 42 52
are u referring to this one? it is actually from the message i reply to!

**Pull request merged**